packages/core/src/Stores/client-store.js

```diff
@@ -1226,9 +1226,23 @@ export default class ClientStore extends BaseStore {
         const search = window.location.search;
         const search_params = new URLSearchParams(search);
         const redirect_url = search_params?.get('redirect_url');
+        const code_param = search_params?.get('code');
+        const action_param = search_params?.get('action');
 
         this.setIsLoggingIn(true);
         const authorize_response = await this.setUserLogin(login_new_user);
+
+        if (search) {
+            if (code_param && action_param) this.setVerificationCode(code_param, action_param);
+            document.addEventListener('DOMContentLoaded', () => {
+                setTimeout(() => {
+                    // timeout is needed to get the token (code) from the URL before we hide it from the URL
+                    // and from LiveChat that gets the URL from Window, particularly when initialized via HTML script on mobile
+                    history.replaceState(null, null, window.location.search.replace(/&?code=[^&]*/i, ''));
```

Comment:
since query params is seperated by `&`, the `&` before code is always existed, so why do we need to make it optional?
```suggestion
                    history.replaceState(null, null, window.location.search.replace(/&code=[^&]*/i, ''));
```

Reply:
@mahdiyeh-fs because we need to include a case if the `code` query would be the first one on the list of query params, i.e. right after the '?' in the url 🙂